scripts/core/EXP-COMMAND-BUFFER.rst

```diff
@@ -237,6 +282,71 @@ parameters to the kernel and the execution ND-Range.
     // Perform the update
     ${x}CommandBufferUpdateKernelLaunchExp(hCommand, &update);
 
+Command Event Update
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+Once a command-buffer has been finalized the wait-list parameter of the command
+can be updated with ${x}CommandBufferUpdateWaitEventsExp. The number of wait
+events for a command must stay consistent, therefore the number of events
+passed to ${x}CommandBufferUpdateWaitEventsExp must be the same as when the
+command was created.
+
+The ${x}CommandBufferUpdateSignalEventExp entry-points can be used to update
+the signal event of a command. This returns a new event that will be signaled
```

Comment:
In fact maybe there is a deeper problem with the `get_event()` changes. If we want a UR event which only represents the most recent execution of a graph we have the same problem as SYCL: they are one-shot events. So we basically need a way to get a new UR event (which will have its status reset) which is backed by the same backend adapter event which is reset automatically, to prevent having to update the command-buffer.

I think we can accomplish this in level zero already it looks like, since UR event handle can be non-owning when it comes to the underlying L0 event.

Reply:
I've modified the implementation to use the same CUDA event object to back the new ur events that are returned in a non-owning way.